src/DropdownMenu.js

```diff
@@ -41,7 +41,23 @@ const DropdownMenu = (props, context) => {
 
   if (context.isOpen && !context.inNavbar) {
     Tag = Popper;
-    const position1 = context.dropup ? 'top' : 'bottom';
+
+    let position1;
+    switch (context.direction) {
```

Comment:
You can use a hashmap instead and it will be more performant and less code.
```
// this defined outside of the component:
const directionPositionMap = {
  up: 'top',
  left: 'left',
  right: 'right',
  down: 'bottom',
};
// this right here:
const position1 = directionPositionMap[context.direction] || 'bottom';
```
You don't really need the `|| 'bottom'` as the contextTypes should ensure it is in the hashmap.... but just in case?